### PR TITLE
Adding Blocks Tablet App to Marketplace

### PR DESCRIPTION
**Pull request by elisa-lj11**
**TEST PLAN**

1. Blocks app should load as default app (named "BLOCKS" with geometric shape icon)
2. Open "BLOCKS"
3. Click on any model
4. New page should open with model preview and "Download" button (if "Download" doesn't appear, that means the model is not remixable, so you'll have to right-click to go back to the item listings if you are in desktop mode or click the "Back" button if you are in HMD mode)
5. Click "Download"
6. Model should appear in front of you with colors

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2366/HighFidelity-Beta-PR11147-17acf0520d5e682f6d4d666deb5f050cf26511f2.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2366/HighFidelity-Beta-PR11147-17acf0520d5e682f6d4d666deb5f050cf26511f2.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2385/HighFidelity-Beta-PR11147-74c90d632380c85fb24c4cd12b75a05ecf9b01b9.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2395/HighFidelity-Beta-PR11147-d3c76c99c20ab8598dfea5ced9487b9920f95c2b.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2395/HighFidelity-Beta-PR11147-d3c76c99c20ab8598dfea5ced9487b9920f95c2b.dmg

**Comment by elisa-lj11** [comment]
**TEST PLAN**

1. Blocks app should load as default app (named "BLOCKS" with geometric shape icon)
2. Open "BLOCKS"
3. Click on any model
4. New page should open with model preview and "Download" button
5. Click "Download"
6. Model should appear in front of you with colors

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2428/HighFidelity-Beta-PR11147-28476960bb3d6560e941e260c99e2a9b91ca9f73.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2428/HighFidelity-Beta-PR11147-28476960bb3d6560e941e260c99e2a9b91ca9f73.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2436/HighFidelity-Beta-PR11147-da5bb03187431111156690788e17ef848a6989d4.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2439/HighFidelity-Beta-PR11147-6c2bf633929b86cd723c26ea9d411706782e9ce3.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2439/HighFidelity-Beta-PR11147-6c2bf633929b86cd723c26ea9d411706782e9ce3.exe

**Comment by Nex-Pro** [comment]
testing

**Comment by Nex-Pro** [comment]
Icon is broken in toolbar and tablet mode. Graphic is black. Clearing cache did not fix. Should it be white ?

### TEST PLAN

Blocks app should load as default app (named "BLOCKS" with geometric shape icon) **SUCCESS**
Open "BLOCKS"
Click on any model
New page should open with model preview and "Download" button (if "Download" doesn't appear, that means the model is not remixable, so you'll have to right-click to go back to the item listings if you are in desktop mode or click the "Back" button if you are in HMD mode) **SUCCESS**
Click "Download"
Model should appear in front of you with colors **SUCCESS**

Pass apart from issue with Icon colour of graphic.

**Comment by elisa-lj11** [comment]
I used an icon from a tablet app example for testing, but it definitely should be changed. Not sure who designs app icons.

**Comment by Nex-Pro** [comment]
@elisa-lj11 Code reviewer might want a newline at the end of blockApp.js

**Comment by Nex-Pro** [comment]
@elisa-lj11 Nice job.

**Comment by elisa-lj11** [comment]
@Nex-Pro Thank you :) will add the newline when I'm back on Monday

File: libraries/script-engine/src/FileScriptingInterface.cpp [hunk 1]
```diff
@@ -52,14 +53,18 @@ void FileScriptingInterface::runUnzip(QString path, QUrl url, bool autoAdd, bool
     }
 
     QStringList fileList = unzipFile(path, tempDir);
-    QString filename = QUrl::fromLocalFile(fileList.first()).toString();
+    //QString filename = QUrl::fromLocalFile(fileList.first()).toString();
```
Comment (druiz17) [inline libraries/script-engine/src/FileScriptingInterface.cpp]: remove commented out code

File: scripts/defaultScripts.js [hunk 2]
```diff
@@ -24,6 +24,7 @@ var DEFAULT_SCRIPTS_COMBINED = [
     "system/makeUserConnection.js",
     "system/tablet-goto.js",
     "system/marketplaces/marketplaces.js",
+    "system/marketplaces/blocksApp.js", // temporarily in default scripts for testing purposes
```
Comment (druiz17) [inline scripts/defaultScripts.js]: remove blocksApp.js from the defaultScripts.

**Comment by themelissabrown** [comment]
@elisa-lj11 let's have Alan make an icon for this

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2499/HighFidelity-Beta-PR11147-072f3dfb03a203ac546acf631f5bdb484d1213c0.exe

**Comment by elisa-lj11** [comment]
@druiz17 changes have been made

**Comment by tapalisa** [comment]
I have one person who does not have their tester flag set running this PR. Please check your environment variables all.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2577/HighFidelity-Beta-PR11147-2b44608ece938cab741a3f8a80a374d472d84233.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2577/HighFidelity-Beta-PR11147-2b44608ece938cab741a3f8a80a374d472d84233.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2595/HighFidelity-Beta-PR11147-4c28b9f09a5550c722fad1304014e00623765197.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2595/HighFidelity-Beta-PR11147-4c28b9f09a5550c722fad1304014e00623765197.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2603/HighFidelity-Beta-PR11147-762f1a9c5c2d1f99745f94371cf0565e6889ad92.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2603/HighFidelity-Beta-PR11147-762f1a9c5c2d1f99745f94371cf0565e6889ad92.exe

**Comment by ShoaibSmith** [comment]
Testing

**Comment by ShoaibSmith** [comment]
I've been having little problems with this build. At first HMD mode was not working. It would go in and out of HMD mode, then it would just freeze. After I uninstalled and re installed, it worked properly. I continued on with the tests but was not able to get the blocks app running. I ran the scripts to try and see if it would work, but it didn't. I then tried to uninstall and re install again to see if it worked properly. It did not. Am I possibly missing something?

**Comment by misslivirose** [comment]
@elisa-lj11 I am able to run everything so I would say that this is QA passed, but there are no styling updates applied to the "Updated required" page. This should match our default styling - once that's fixed then I'd say this is good to go.

**Comment by elisa-lj11** [comment]
Styling has been updated: https://hifi-content.s3.amazonaws.com/elisalj/blocks/updateToBlocks.html

**Pull request merged**